test/test_drghmc.py

```diff
@@ -0,0 +1,144 @@
+from test.models.binomial import Binomial
```

Comment:
You have defined a ton of combinatorics in your top level function and here's where it comes home to roost.  You need to test **all** of the combinatorics of your arguments.  And you want to test all of the options.

You want to test all your exceptional cases are throwing the right exceptions.

You want to test with probabilistic delayed rejection turned on and with it turned off.

If you cut down on the API flexibility, you can also cut down on the combinatorics of testing.  You should always be designing functions/doc/tests together.  If any of them feels "hard", it's a bad sign that the design is off.